api/main.py

```diff
@@ -466,6 +466,271 @@ def strip_ansi_codes(text):
+                                name = agent_data['metadata']['name']
+                                description = agent_data.get('spec', {}).get('description', '')
+                                agents_info.append({
+                                    'name': name,
+                                    'description': description
+                                })
+                except yaml.YAMLError:
+                    name_matches = re.findall(r'name:\s*(\w+)', agents_yaml)
+                    desc_matches = re.findall(r'description:\s*\|\s*\n\s*(.+?)(?=\n\s*\w+:|$)', agents_yaml, re.DOTALL)
+                    
+                    for i, name in enumerate(name_matches):
+                        description = desc_matches[i] if i < len(desc_matches) else ""
+                        agents_info.append({
+                            'name': name,
+                            'description': description.strip()
+                        })
+
+                workflow_prompt = f"Create a workflow that uses the following agents:\n\n"
+                for i, agent in enumerate(agents_info, 1):
+                    workflow_prompt += f"agent{i}: {agent['name']} – {agent['description']}\n"
+                workflow_prompt += f"\nprompt: {request.content}"
+
+                print("Calling workflow generation on port 8004...")
+                workflow_resp = requests.post(
+                    "http://localhost:8004/chat",
+                    json={"prompt": workflow_prompt, "agent": "WorkflowYAMLBuilder"},
+                    timeout=120
+                )
+                print(f"Workflow generation response status: {workflow_resp.status_code}")
+                if workflow_resp.status_code != 200:
+                    print(f"Workflow generation error: {workflow_resp.text}")
+                    raise Exception(f"Workflow generation failed: {workflow_resp.text}")
+
+                workflow_output = workflow_resp.json().get("response", "")
+                print(f"Workflow generation output length: {len(workflow_output)}")
+                workflow_yaml = ""
+                if "```yaml" in workflow_output:
+                    workflow_yaml = (
+                        workflow_output.split("```yaml", 1)[-1].split("```", 1)[0].strip()
+                    )
+                elif "```" in workflow_output:
+                    workflow_yaml = workflow_output.split("```", 1)[-1].split("```", 1)[0].strip()
+                else:
+                    yaml_match = re.search(r'apiVersion:.*?(?=\n\n|\Z)', workflow_output, re.DOTALL)
+                    if yaml_match:
+                        workflow_yaml = yaml_match.group(0).strip()
+                
+                print(f"Extracted workflow YAML length: {len(workflow_yaml)}")
+
+                clean_response = f"""✅ Successfully generated both agents.yaml and workflow.yaml from your prompt!
```

Comment:
This clearly could be refactored into a ClasifyUserRequest with various private methods and a classify(query) -> string (or some enumeration) so that this code is not just one massive if else if ... which will be super hard to maintain